code/modules/holomap/holomap.dm

```diff
@@ -0,0 +1,233 @@
+/*
+	Somone should totally add an action button to select a location when observing the map
+	and it display a list to select from (TGUI maybe!? color coded !? YES?)
+	then show an icon like YouAreHere there
+*/
+
+/obj/machinery/holomap
+	name = "Holomap"
+	desc = "It displays a holographic map of the station."
+	icon = 'icons/obj/terminals.dmi'
+	icon_state = "map"
+	anchored = TRUE
+	idle_power_consumption = 5
+	active_power_consumption = 15
+	var/display_map //for mappers who want to add holomaps for off station locations, otherwise it gets set in Initialize()
+	var/list/display_targets = list()
+	var/image/holomap_projection
+	var/image/YouAreHere
+
+/obj/machinery/holomap/Initialize(mapload)
+	. = ..()
+	if(!display_map)
+		display_map = SSmapping.map_datum.technical_name
+	var/icon/base_icon = make_map()
```

Comment:
This should probably be called only once in the entire session. Cache `display_map` and check if it exists already before generating it pointlessly

Reply:
Yes i totally agree

Reply:
> This should probably be called only once in the entire session. Cache `display_map` and check if it exists already before generating it pointlessly

It would be so expensive... but it would be cool if it updated every 10 minutes or so. The SM blew up and engineering is gone? Giant crater there